File: client/src/assets/song.py
class Song:
    def __init__(self, title):
        
        # song title used for lookup in dictionary
        self.title = title;
        
        # next node in regular-order doubly linked list
        self.below = set();

        # next node in reverse-order doubly linked list
        self.above = set();

    def create_below_edge(self, song_node):

        if(song_node == self):
            pass;
        else:
            self.below.add(song_node);

    def create_above_edge(self, song_node):

        if(song_node == self):
            pass
        else:
            self.above.add(song_node);
    

    def check_below(self):
        return (len(self.below) > 1);

    def check_above(self):
        return (len(self.above) > 1);

    def get_below(self):
        return self.below
    
    def get_above(self):
        return self.above
    
    def get_title(self):
        return self.title



 

File: client/src/Routes/RankingPage/song.js

class Song {
    constructor(title) {
        // Song title used for lookup in hashtable. 
        this.title = title,

        // Linked references to child song nodes. 
        this.below = new Set()
    }

    // Create an edge to a new child song node. 
    create_below_edge = (song_node) => {

        if (song_node != this) {
            this.below.add(song_node);
        }

    };

    // Remove an edge between this node and an input node, if it exists.
    remove_edge = (song_node) => {
        this.below.delete(song_node);
    };

    // Return child nodes. 
    get_below = () => {
        return this.below;
    };

    // Return the title of this song node. 
    get_title = () => {
        return this.title;
    };

}